include/os/linux/zfs/sys/zfs_znode_impl.h

```diff
@@ -73,7 +73,14 @@ extern "C" {
 #define	zn_has_cached_data(zp)		((zp)->z_is_mapped)
 #define	zn_rlimit_fsize(zp, uio)	(0)
 
+#if defined(__GNUC__)
+/* it's critical that callers check, VERIFY(), or explicitly discard the return of igrab() and thus zhold(); a NULL return means that a reference was not added so a symmetric zrele() must not be performed */
+static inline struct inode * __attribute__((warn_unused_result))
```

Comment:
So it looks like clang supports `warn_unused_result` as well, so maybe there's no need for the `__GNUC__` wrapper?  Or perhaps there's some way we can allow it for both gcc and clang?

nit: 80 characters column limit.

Reply:
clang sensibly defines __GNUC__ too!